### PR TITLE
implement train chunk and prepare for nlu components except extractors and classifiers

### PR DESCRIPTION
**Pull request by Ghostvv**
**Proposed changes**:
- Closes https://github.com/RasaHQ/rasa/issues/7274
- Closes https://github.com/RasaHQ/rasa/issues/7277

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by Ghostvv** [comment]
@tabergma since it moving code around, not sure if I should any tests?

File: rasa/nlu/classifiers/keyword_intent_classifier.py
Comment (tabergma) [inline rasa/nlu/classifiers/keyword_intent_classifier.py:51]: Would not raise `Exception` here but rather a more specific one. What about creating a `RasaTrainException` in https://github.com/RasaHQ/rasa/blob/master/rasa/shared/exceptions.py, subclassing our `RasaException`?

**Review by tabergma — COMMENTED** [review]
Looks great 💯 Just left one comment around the exception.

Regarding tests: We could maybe add a tests for tokenizers and maybe some of the classifiers, that checks if `train_chunk` raises an exception. And maybe a test that checks that `prepare_partial_training` is not modifying the training data.

**Comment by Ghostvv** [comment]
> And maybe a test that checks that prepare_partial_training is not modifying the training data.

how can I check that?

**Comment by tabergma** [comment]
Daksh implement a test like this [in his PR](https://github.com/RasaHQ/rasa/pull/7352/files#diff-f1bd9df57cdd694505dea844b8abfca6447e62187de92fa7a56d1f4623b9110bR642).

File: rasa/nlu/classifiers/keyword_intent_classifier.py
Comment (Ghostvv) [reply rasa/nlu/classifiers/keyword_intent_classifier.py:51]: created `RasaTrainChunkException`

File: rasa/shared/nlu/training_data/training_data.py
Comment (tabergma) [inline rasa/shared/nlu/training_data/training_data.py:723]: I think we can remove those again as `TrainingDataChunk` now inherits from `TrainingData` once you merge `partial-dataset-loading` branch into yours.

**Review by tabergma — APPROVED** [review]
Thanks for adding tests 💯

**Comment by Ghostvv** [comment]
@tabergma @dakshvar22 the tests fail due to timeout on `LanguageModelFeaturizer` and `HFTransformersNLP` should I exclude them form the test?

**Comment by tabergma** [comment]
Yeah, I think Felicia is looking into it. But let's add a TODO, so that we don't forget about those. (And I think you can squash merge right after you update the code, don't think you need to wait until the tests pass again as everything else worked.)

**Pull request merged**
